core/arch/arm32/tee/entry.c

```diff
@@ -317,6 +317,26 @@ static void entry_cancel(struct thread_smc_args *args,
 }
 
 
+#if (CFG_MEASURE_PERFORMANCE == 1)
+
+static inline void print_cntvct_with_message(uint32_t cmd, const char *msg)
```

Comment:
IMO this is too specific. I would rather integrate this with the tracing macros. Say: if CFG_TRACE_TIMESTAMPS is 'y', then DMSG/IMSG/EMSG call tee_svc_get_sys_counter and prepend the counter to the trace.